src/solver/conv_hip_implicit_gemm_mlir_cpp_fwd.cpp

```diff
@@ -0,0 +1,120 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <miopen/conv/invokers/impl_gemm.hpp>
+#include <miopen/solver.hpp>
+#include <miopen/handle.hpp>
+#include <miopen/generic_search.hpp>
+
+#include "implicitgemm_util.hpp"
+
+namespace miopen {
+namespace solver {
+
+bool ConvHipImplicitGemmMlirCppFwd::IsApplicable(const ConvolutionContext& ctx) const
+{
+#ifdef MIOPEN_LIBMLIRMIOPEN
+    if(ctx.Is3d())
+        return false;
+    return ConvHipImplicitGemmV4R4Fwd::IsApplicable(ctx);
+#else
+    std::ignore = ctx;
+    return false;
+#endif
+}
+
+PerformanceImplicitGemmMlirCppFwd
+ConvHipImplicitGemmMlirCppFwd::GetPerformanceConfig(const ConvolutionContext& ctx) const
+{
+    return GetPerformanceConfigBase<PerformanceImplicitGemmMlirCppFwd>(ctx);
+}
+
+PerformanceImplicitGemmMlirCppFwd
+ConvHipImplicitGemmMlirCppFwd::Search(const ConvolutionContext& context,
+                                      const AnyInvokeParams& invoke_ctx) const
+{
+    return GenericSearch(*this, context, invoke_ctx);
+}
+
+ConvSolution ConvHipImplicitGemmMlirCppFwd::GetSolution(
+    const ConvolutionContext& ctx, const PerformanceImplicitGemmMlirCppFwd& config, bool) const
+{
+    ConvSolution result;
+    KernelInfo construction_parameters;
+
+    assert(config.IsValid(ctx));
+
+    int grid_size = 0;
+
+    std::tie(grid_size, std::ignore) = config.CalculateGridSize(ctx);
+
+    construction_parameters.l_wk.push_back(config.BlockSize);
+    construction_parameters.l_wk.push_back(1);
+    construction_parameters.l_wk.push_back(1);
+
+    construction_parameters.g_wk.push_back(config.BlockSize * grid_size);
+    construction_parameters.g_wk.push_back(1);
+    construction_parameters.g_wk.push_back(1);
+
+    std::string version   = "v4r4";
+    std::string direction = "fwd";
+    std::string operation = "conv2d";
+
+    construction_parameters.kernel_file =
+        "mlir_gen_igemm_conv2d_cpp_" + version + "_" + direction + ".cpp";
+
+    construction_parameters.kernel_name = "mlir_gen_igemm_conv2d_cpp_" + version + "_" + direction;
+
+    // Arguments for mlir-miopen-driver.
+    using CI = ConvolutionContextInterpreter;
+    construction_parameters.comp_options =
+        std::string(" --operation ") + operation + std::string(" --fil_layout ") +
+        CI::GetFilterLayout(ctx) + std::string(" --in_layout ") + CI::GetInputLayout(ctx) +
```

Comment:
Preferable to use a string stream instead of concatenating strings

Reply:
Sure

Reply:
I double checked other solvers, in particular `conv_hip_implicit_gemm_fwd_v4r4.cpp`. It is following the same use convention as in there. Marking it as resolved for now.